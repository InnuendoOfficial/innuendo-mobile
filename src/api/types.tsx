// types.tsx
/**
 * Store the api types.
 */

export {}