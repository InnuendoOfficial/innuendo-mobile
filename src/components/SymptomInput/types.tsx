type SymptomInputProps = {
  title: string,
  description: string,
  value: number | boolean,
  onValueChange?: (value: number) => void
}

export type {
  SymptomInputProps
}