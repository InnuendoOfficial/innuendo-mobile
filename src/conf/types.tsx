import { ImageSourcePropType } from "react-native"
import Report from '../types';

type Symptom = {
  reportKey?: string,
  input: {
    title: string,
    indicator?: string,
  },
  display: string | ((report: Report) => string),
}

type SymptomCategory = {
  name: string,
  icon: ImageSourcePropType,
  image: ImageSourcePropType,
  inputIndicator: string,
  isCompleted: (report: Report) => boolean,
  field: keyof Report,
  symptoms: Symptom[]
}

export type { Symptom, SymptomCategory }