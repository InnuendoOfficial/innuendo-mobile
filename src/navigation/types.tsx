import { NativeStackNavigationProp } from "@react-navigation/native-stack";
import { BottomTabNavigationProp } from "@react-navigation/bottom-tabs";

type TabParamList = {
  Home: undefined;
  Calendar: undefined;
  ShareReport: undefined | { accessCode: string };
};

type StackParamList = {
  AppIntro: undefined;
  Splash: undefined;
  Login: undefined;
  SignUp: undefined;
  Tabs: undefined;
  Endoscore: undefined;
  Settings: undefined;
  Feedback: undefined;
  DeleteAccount: undefined;
  ChangePassword: undefined;
  ChangeEmail: undefined;
  DeleteData: undefined;
  ViewReport: undefined;
  EditReport: undefined;
  EditReportSymptom: { symptomName: string };
  ShareReportChooseSymptoms: undefined;
};

type StackNavProp = NativeStackNavigationProp<StackParamList>;
type TabNavProp = BottomTabNavigationProp<TabParamList>;

export type { TabParamList, StackParamList, StackNavProp, TabNavProp };
