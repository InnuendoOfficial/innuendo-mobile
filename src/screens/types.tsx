import { NativeStackScreenProps } from '@react-navigation/native-stack';
import { BottomTabScreenProps } from '@react-navigation/bottom-tabs';
import { CompositeScreenProps } from '@react-navigation/native';
import { StackParamList, TabParamList } from '../navigation/types';

type EditReportProps = CompositeScreenProps<
  NativeStackScreenProps<StackParamList, 'EditReport'>,
  BottomTabScreenProps<TabParamList>
>

type EditReportSymptomProps = CompositeScreenProps<
  NativeStackScreenProps<StackParamList, 'EditReportSymptom'>,
  BottomTabScreenProps<TabParamList>
>

type SettingsScreenProps = NativeStackScreenProps<StackParamList, 'Settings'>
type FeedbackScreenProps = NativeStackScreenProps<StackParamList, 'Feedback'>

export type {
  EditReportProps,
  EditReportSymptomProps,
  SettingsScreenProps,
  FeedbackScreenProps
}