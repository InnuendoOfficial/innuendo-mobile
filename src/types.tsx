import React from 'react'

type Period = {
  pain: number,
  flow: number
}

type DigestiveDisorders = {
  pain: number, // from 1 to 10
  presenceOfBlood: boolean,
  diarrhea: boolean,
  constipation: boolean
}

type UrinaryDisorders = {
  pain: number,
  burn: boolean,
  presenceOfBlood: boolean,
  troubleEmptyingBladder: boolean,
  frequentUrges: boolean
}

type Fatigue = {
  fatigue: boolean,
  lowerBackPain: boolean,
  pelvicPain: boolean,
}

type Report = {
  date: string, // format: "YYYY-MM-DD", ex: "2021-12-14"
  period: Period,
  digestiveDisorders: DigestiveDisorders,
  urinaryDisorders: UrinaryDisorders,
  fatigue: Fatigue,
  painDuringSex: boolean,
  medicines: string[]
}

type ReportStateProps = {
  report: Report,
  setReport: React.Dispatch<React.SetStateAction<Report>>
}

const blankReport: Report = {
  date: "",
  period: {
    pain: 0,
    flow: 0
  },
  digestiveDisorders: {
    pain: 0,
    presenceOfBlood: false,
    diarrhea: false,
    constipation: false
  },
  urinaryDisorders: {
    pain: 0,
    burn: false,
    presenceOfBlood: false,
    troubleEmptyingBladder: false,
    frequentUrges: false
  },
  fatigue: {
    fatigue: false,
    lowerBackPain: false,
    pelvicPain: false,
  },
  painDuringSex: false,
  medicines: [""]
}

export { blankReport }
export type { ReportStateProps }
export default Report