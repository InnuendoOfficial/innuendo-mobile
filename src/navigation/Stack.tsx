import React from 'react'
import { DarkTheme, DefaultTheme, NavigationContainer } from '@react-navigation/native';
import { createNativeStackNavigator, NativeStackNavigationOptions } from '@react-navigation/native-stack';
import { StackParamList } from './types'
import { navigationTheme } from '../theme';

import LoginScreen from '../screens/auth/Login';
import SignUpScreen from '../screens/auth/SignUp';
import TabsNavigation from './Tabs';
import EditReportScreen from '../screens/EditReport';
import EditReportCategoryScreen from '../screens/EditReportCategory';
import ShareReportChooseSymptomsScreen from '../screens/ShareReportChooseSymptoms';
import { useColorModeValue } from 'native-base';
import Logo from '../components/NavHeader/Logo';

const Stack = createNativeStackNavigator<StackParamList>();

function StackNavigation() {
  return (
    <NavigationContainer theme={useColorModeValue(DefaultTheme, DarkTheme)}>
      <Stack.Navigator screenOptions={options}>
        {/* <Stack.Screen name="Login" component={LoginScreen} options={{ headerShown: false }}/>
        <Stack.Screen name="SignUp" component={SignUpScreen} options={{ headerShown: false }}/> */}
        <Stack.Screen name="Tabs" component={TabsNavigation} options={{ headerShown: false }}/>
        <Stack.Screen name="EditReport" component={EditReportScreen}/>
        <Stack.Screen name="EditReportCategory" component={EditReportCategoryScreen}/>
        <Stack.Screen name="ShareReportChooseSymptoms" component={ShareReportChooseSymptomsScreen} />
      </Stack.Navigator>
    </NavigationContainer>
  );
}

const options: NativeStackNavigationOptions = {
  headerTintColor: 'white', // White go back button
  headerTitle: () => <Logo />,
  headerTitleAlign: "center",
  headerStyle: {
    backgroundColor: "#776CCB"
  },
  headerBackTitleVisible: false
}

export default StackNavigation