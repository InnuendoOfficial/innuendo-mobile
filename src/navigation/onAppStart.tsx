import { AuthState } from "../types/auth";
import { StorageManager, ColorMode } from "native-base";
import api from "../api";
import { isAppIntroPassed, retrieveUserSessionFromStorage, } from "../storage";

async function retrieveAuthFromStorage(): Promise<AuthState> {
  const introIsPassed = await isAppIntroPassed()
  const session = await retrieveUserSessionFromStorage()

  api.tokens.setAccessToken(session?.access_token || "")
  return ({
    isLoading: false,
    isSignedIn: (session !== undefined),
    isFirstTimeUsingApp: (!introIsPassed),
  })
};

export { retrieveAuthFromStorage }