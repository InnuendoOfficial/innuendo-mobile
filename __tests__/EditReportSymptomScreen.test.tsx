import React from 'react';
import { render, waitFor, fireEvent } from '@testing-library/react-native';
import { QueryClient, QueryClientProvider } from 'react-query';
import { NativeBaseProvider } from 'native-base';
import { NavigationContainer } from '@react-navigation/native';
import { createNativeStackNavigator } from '@react-navigation/native-stack';
import EditReportSymptomScreen from '../src/screens/EditReportSymptom';
import { StackParamList } from '../src/navigation/types';
type NavigationScreenPropAlias = NavigationScreenProp<{}>;

const inset = {
  frame: { x: 0, y: 0, width: 0, height: 0 },
  insets: { top: 0, left: 0, right: 0, bottom: 0 },
};
const Stack = createNativeStackNavigator<StackParamList>();

describe("EditReportSymptomScreen", () => {
  it('renders screen with data from API', async () => {
    const queryClient = new QueryClient({
      defaultOptions: {
        queries: {
          retry: false,
          cacheTime: Infinity
        }
      }
    })
    const tree = render(
      <NativeBaseProvider initialWindowMetrics={inset}>
        <QueryClientProvider client={queryClient}>
          <NavigationContainer>
            <Stack.Navigator>
              <Stack.Screen
                name="EditReportSymptom"
                component={EditReportSymptomScreen}
                initialParams={{ symptomName: "douleur menstruelle"}}
              />
            </Stack.Navigator>
          </NavigationContainer>
        </QueryClientProvider>
      </NativeBaseProvider>
    );

    await waitFor(() => expect(tree.getByLabelText("loading")).toBeDefined());
    await waitFor(() => expect(tree.getByText("douleur menstruelle")).toBeDefined());
    expect(tree.getByText("Valider")).toBeDefined()
    // expect(tree).toMatchSnapshot()
  });

  it('goes back when the "Go Back" button is pressed', () => {
    const queryClient = new QueryClient({
      defaultOptions: {
        queries: {
          retry: false,
          cacheTime: Infinity
        }
      }
    })

    const tree = render(
      <NativeBaseProvider initialWindowMetrics={inset}>
        <QueryClientProvider client={queryClient}>
          <NavigationContainer>
            <Stack.Navigator>
              <Stack.Screen
                name="EditReportSymptom"
                component={EditReportSymptomScreen}
                initialParams={{
                  symptomName: "douleur menstruelle",
                }}
              />
            </Stack.Navigator>
          </NavigationContainer>
        </QueryClientProvider>
      </NativeBaseProvider>
    );

    const goBackButton = tree.getByText("Valider")
    expect(goBackButton).toBeDefined()
    fireEvent.press(goBackButton)
    // expect(goBackButton).toThrowError()
  });
})