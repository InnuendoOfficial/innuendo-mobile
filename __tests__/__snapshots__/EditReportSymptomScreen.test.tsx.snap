// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditReportSymptomScreen renders screen with data from API 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "flex": 1,
        },
        undefined,
      ]
    }
  >
    <RNSScreenStack
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <RNSScreen
        collapsable={false}
        forwardedRef={[Function]}
        nativeID="animatedComponent"
        onAppear={[Function]}
        onDisappear={[Function]}
        onDismissed={[Function]}
        onWillDisappear={[Function]}
        replaceAnimation="push"
        stackPresentation="push"
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <RNSScreenStackHeaderConfig
          backButtonInCustomView={false}
          backgroundColor="rgb(255, 255, 255)"
          color="rgb(0, 122, 255)"
          direction="ltr"
          disableBackButtonMenu={false}
          hidden={false}
          hideBackButton={false}
          hideShadow={false}
          largeTitleHideShadow={false}
          title="EditReportSymptom"
          titleColor="rgb(28, 28, 30)"
          topInsetEnabled={false}
          translucent={false}
        />
        <View
          style={
            Array [
              Object {
                "flex": 1,
              },
              Object {
                "backgroundColor": "rgb(242, 242, 242)",
              },
              undefined,
            ]
          }
        >
          <View
            dataSet={Object {}}
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "paddingBottom": 20,
                "paddingTop": 20,
              }
            }
          >
            <View
              dataSet={Object {}}
              style={
                Object {
                  "alignItems": "center",
                  "flex": 1,
                  "width": "90%",
                }
              }
            >
              <View
                dataSet={Object {}}
                style={
                  Object {
                    "borderRadius": 10,
                    "paddingBottom": 8,
                    "paddingLeft": 16,
                    "paddingRight": 16,
                    "paddingTop": 8,
                  }
                }
              >
                <View
                  dataSet={Object {}}
                  style={
                    Object {
                      "alignItems": "center",
                    }
                  }
                >
                  <Text
                    dataSet={Object {}}
                    style={
                      Object {
                        "backgroundColor": undefined,
                        "color": "#171717",
                        "fontFamily": undefined,
                        "fontSize": 18,
                        "fontStyle": "normal",
                        "fontWeight": "700",
                        "letterSpacing": 0,
                        "lineHeight": 27,
                        "textDecorationLine": undefined,
                      }
                    }
                  >
                    douleur menstruelle
                  </Text>
                  <View
                    dataSet={Object {}}
                    style={
                      Object {
                        "height": 8,
                      }
                    }
                  />
                  <View
                    _interactionBox={
                      Object {
                        "p": "2",
                      }
                    }
                    aria-label="Slider"
                    dataSet={Object {}}
                    maxValue={10}
                    minValue={0}
                    onChange={[Function]}
                    sliderTrackHeight={4}
                    style={
                      Object {
                        "alignItems": "center",
                        "height": 50,
                        "justifyContent": "center",
                        "marginBottom": -16,
                        "marginTop": -16,
                        "width": 300,
                      }
                    }
                    thumbSize={4}
                    value={
                      Array [
                        0,
                      ]
                    }
                  >
                    <View
                      accessible={true}
                      collapsable={false}
                      dataSet={Object {}}
                      focusable={true}
                      onBlur={[Function]}
                      onClick={[Function]}
                      onFocus={[Function]}
                      onHoverIn={[Function]}
                      onHoverOut={[Function]}
                      onLayout={[Function]}
                      onResponderGrant={[Function]}
                      onResponderMove={[Function]}
                      onResponderRelease={[Function]}
                      onResponderTerminate={[Function]}
                      onResponderTerminationRequest={[Function]}
                      onStartShouldSetResponder={[Function]}
                      style={
                        Object {
                          "alignItems": "center",
                          "height": 16,
                          "justifyContent": "center",
                          "paddingBottom": 12,
                          "paddingTop": 12,
                          "width": "100%",
                        }
                      }
                    >
                      <View
                        dataSet={Object {}}
                        isVertical={false}
                        style={
                          Array [
                            Object {
                              "backgroundColor": "#e5e5e5",
                              "borderRadius": 8,
                              "height": 16,
                              "overflow": "hidden",
                              "width": 16,
                            },
                            Object {
                              "height": 4,
                              "width": "100%",
                            },
                          ]
                        }
                      >
                        <View
                          colorScheme="primary"
                          dataSet={Object {}}
                          sliderTrackPosition="0%"
                          style={
                            Array [
                              Object {
                                "backgroundColor": "#0891b2",
                                "height": 16,
                                "left": 0,
                                "position": "absolute",
                                "width": 16,
                              },
                              Object {
                                "height": 4,
                                "width": "0%",
                              },
                            ]
                          }
                        />
                      </View>
                    </View>
                    <View
                      colorScheme="primary"
                      dataSet={Object {}}
                      onBlur={[Function]}
                      onFocus={[Function]}
                      onMoveShouldSetResponder={[Function]}
                      onMoveShouldSetResponderCapture={[Function]}
                      onResponderEnd={[Function]}
                      onResponderGrant={[Function]}
                      onResponderMove={[Function]}
                      onResponderReject={[Function]}
                      onResponderRelease={[Function]}
                      onResponderStart={[Function]}
                      onResponderTerminate={[Function]}
                      onResponderTerminationRequest={[Function]}
                      onStartShouldSetResponder={[Function]}
                      onStartShouldSetResponderCapture={[Function]}
                      scaleOnPressed={1}
                      style={
                        Array [
                          Object {
                            "alignItems": "center",
                            "backgroundColor": "#0891b2",
                            "borderRadius": 9999,
                            "height": 16,
                            "justifyContent": "center",
                            "position": "absolute",
                            "width": 16,
                            "zIndex": 999,
                          },
                          Array [
                            Object {
                              "bottom": undefined,
                              "left": "0%",
                              "transform": Array [
                                Object {
                                  "translateX": -8,
                                },
                                Object {
                                  "scale": 1,
                                },
                              ],
                            },
                            undefined,
                          ],
                        ]
                      }
                    >
                      <View
                        dataSet={Object {}}
                        style={
                          Object {
                            "alignItems": "center",
                            "flexDirection": "row",
                            "justifyContent": "center",
                          }
                        }
                      >
                        <View
                          dataSet={Object {}}
                          style={
                            Object {
                              "borderRadius": 9999,
                              "paddingBottom": 8,
                              "paddingLeft": 8,
                              "paddingRight": 8,
                              "paddingTop": 8,
                              "position": "absolute",
                              "zIndex": -1,
                            }
                          }
                        />
                        <View
                          dataSet={Object {}}
                          style={
                            Object {
                              "width": 8,
                            }
                          }
                        />
                        <View
                          colorScheme="primary"
                          dataSet={Object {}}
                          scaleOnPressed={1}
                          style={
                            Object {
                              "alignItems": "center",
                              "backgroundColor": "#0891b2",
                              "borderRadius": 9999,
                              "display": "flex",
                              "justifyContent": "center",
                            }
                          }
                        />
                      </View>
                    </View>
                  </View>
                  <View
                    dataSet={Object {}}
                    style={
                      Object {
                        "height": 8,
                      }
                    }
                  />
                  <Text
                    dataSet={Object {}}
                    style={
                      Object {
                        "backgroundColor": undefined,
                        "color": "#171717",
                        "fontFamily": undefined,
                        "fontSize": 24,
                        "fontStyle": "normal",
                        "fontWeight": "700",
                        "letterSpacing": 0,
                        "lineHeight": 36,
                        "textDecorationLine": undefined,
                      }
                    }
                  >
                    0
                  </Text>
                </View>
              </View>
              <View
                accessibilityRole="button"
                accessible={true}
                collapsable={false}
                colorScheme="primary"
                dataSet={Object {}}
                focusable={true}
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onHoverIn={[Function]}
                onHoverOut={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "#0891b2",
                    "borderRadius": 4,
                    "flexDirection": "row",
                    "justifyContent": "center",
                    "marginTop": 16,
                    "paddingBottom": 12,
                    "paddingLeft": 12,
                    "paddingRight": 12,
                    "paddingTop": 12,
                    "width": "100%",
                  }
                }
              >
                <View
                  dataSet={Object {}}
                  style={
                    Object {
                      "alignItems": "center",
                      "flexDirection": "row",
                    }
                  }
                  test={true}
                >
                  <View
                    dataSet={Object {}}
                    style={Object {}}
                  >
                    <Text
                      dataSet={Object {}}
                      style={
                        Object {
                          "backgroundColor": undefined,
                          "color": "#fafafa",
                          "fontFamily": undefined,
                          "fontSize": 16,
                          "fontStyle": "normal",
                          "fontWeight": "400",
                          "letterSpacing": 0,
                          "lineHeight": 24,
                          "textDecorationLine": undefined,
                        }
                      }
                    >
                      Valider
                    </Text>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </RNSScreen>
    </RNSScreenStack>
  </View>
</RNCSafeAreaProvider>
`;
